src/index.test.ts

```diff
@@ -63,6 +64,77 @@ describe("voices", () => {
 });
 
 describe("tts.websocket", () => {
+  let phonicWebSocket: PhonicWebSocket;
+
+  /**
+   * Promise that resolves with all messages received from the websocket after a period of no messages. Can be awaited
+   * multiple times.
+   */
+  let allMessagesReceived: Promise<PhonicWebSocketResponseMessage[]> | null =
+    null;
+  const maxIdleTime = 5000; // If we don't receive messages for 5 seconds, we consider that we received all of them.
+
+  const safeGet = (
+    messages: PhonicWebSocketResponseMessage[] | null,
+    idx: number,
+  ) => {
+    if (messages === null) {
+      throw new Error("messages is null");
+    }
+    if (messages[idx] === undefined) {
+      throw new Error(`Expected message at index ${idx} but received less`);
+    }
+    return messages[idx];
+  };
+
+  beforeEach(async () => {
+    const phonic = new Phonic(apiKey, { baseUrl });
+    const { data, error } = await phonic.tts.websocket();
+
+    if (error !== null) {
+      throw new Error("Failed to connect to websocket");
+    }
+
+    phonicWebSocket = data.phonicWebSocket;
+
+    const messages: PhonicWebSocketResponseMessage[] = [];
+    let messagesTimeoutId: NodeJS.Timer | null = null;
+    let allMessagesReceivedResolve:
+      | ((messages: PhonicWebSocketResponseMessage[]) => void)
+      | null = null;
+
+    const createNewPromise = () => {
+      messages.length = 0; // Clear previous messages
+      return new Promise<PhonicWebSocketResponseMessage[]>((resolve) => {
+        allMessagesReceivedResolve = resolve;
+      });
+    };
+
+    allMessagesReceived = createNewPromise();
+
+    phonicWebSocket.onMessage((message) => {
+      messages.push(message);
+
+      if (messagesTimeoutId !== null) {
+        clearTimeout(messagesTimeoutId);
+      }
+
+      messagesTimeoutId = setTimeout(() => {
+        if (allMessagesReceivedResolve === null) {
+          throw new Error("allMessagesReceivedResolve should not be null");
+        }
+        // Pass a copy of messages to the resolver because createNewPromise() will clear it.
+        allMessagesReceivedResolve([...messages]);
+        // Create a new promise for next use
+        allMessagesReceived = createNewPromise();
+      }, maxIdleTime);
+    });
+  });
+
+  afterEach(async () => {
+    phonicWebSocket.close();
+  });
```

Comment:
Let's also clear the timeout here, just in case:
```
if (messagesTimeoutId !== null) {
  clearTimeout(messagesTimeoutId);
}
```

Reply:
`messagesTimeoutId` is defined inside `beforeEach` so it will already be out of scope by the time `afterEach` executes right?

Reply:
Right :)